### PR TITLE
Migrate Site Editor 'settings sidebar' e2e tests to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.

PR migrates `site-editor/settings-sidebar.test.js` e2e tests to Playwright.

## Why?
See #38851.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/site-editor/settings-sidebar.spec.js
```

### CODE REVIEW

File: test/e2e/specs/site-editor/settings-sidebar.spec.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+			const settingsSideber = page.getByRole( 'region', {
+				name: 'Editor settings',
+			} );
+			const templateTitle = settingsSideber.locator(
+				'.edit-site-sidebar-card__title'
+			);
+			const templateDescription = settingsSideber.locator(
+				'.edit-site-sidebar-card__description'
+			);
```
Comment (Mamaduka) [inline test/e2e/specs/site-editor/settings-sidebar.spec.js:60]: Had to use the class locator here.

File: test/e2e/specs/site-editor/settings-sidebar.spec.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+			editor,
+			page,
+		} ) => {
+			await editor.openDocumentSettingsSidebar();
+
+			await expect(
+				page
+					.getByRole( 'region', { name: 'Editor settings' } )
+					.getByRole( 'button', { name: 'Template (selected)' } )
+			).toHaveClass( /is-active/ );
```
Comment (Mamaduka) [inline test/e2e/specs/site-editor/settings-sidebar.spec.js:42]: I think we can improve active tab assertions after #56959 is merged.

Untested example:

```js
await expect(
		page
			.getByRole( 'region', { name: 'Editor settings' } )
			.getByRole( 'tab', { selected: true } )
	).toHaveText( 'Template' );
```

**Review by t-hamano — APPROVED** [review]
LGTM 👍In debug mode, I confirmed that all tests are working as intended.

**Comment by Mamaduka** [comment]
Thank you, @t-hamano!

File: test/e2e/specs/site-editor/settings-sidebar.spec.js (hunk 2 above)
Comment (Mamaduka) [reply test/e2e/specs/site-editor/settings-sidebar.spec.js:42]: The active tab assertions work like a charm - https://github.com/WordPress/gutenberg/pull/57446/files.

**Pull request merged**